viewer/src/components/items/Table.js

```diff
@@ -1,100 +1,149 @@
-/* eslint-disable react/jsx-pascal-case */
 import Loading from "../Loading";
 import React, { useEffect, useState } from "react";
-import { tableDataFromCohortData, tableColumnsWithDot, tableColumnsWithUnderscores } from '../../models/Table'
-import { createTheme, ThemeProvider } from '@mui/material';
+import { tableDataFromCohortData, tableColumnsWithDot, tableColumnsWithUnderscores } from '../../models/Table';
+import { createTheme, ThemeProvider, Box, Button } from '@mui/material';
 import { useTracesData } from "../../hooks/useTracesData";
 import { ItemContainer } from "./ItemContainer";
 import CohortSelect from "../select/CohortSelect";
+/* eslint-disable react/jsx-pascal-case */
 import {
     MRT_ShowHideColumnsButton,
     MRT_TablePagination,
     MRT_ToggleDensePaddingButton,
     MRT_ToggleFiltersButton,
     MRT_ToolbarAlertBanner,
+    MRT_GlobalFilterTextField,
     useMaterialReactTable,
     MRT_TableContainer,
 } from 'material-react-table';
-import { Box } from '@mui/material';
-
+/* eslint-enable react/jsx-pascal-case */
+import FileDownloadIcon from '@mui/icons-material/FileDownload';
+import { mkConfig, generateCsv } from "export-to-csv";
 
 const Table = ({ table, project, itemWidth, height, width }) => {
-    const traceNames = table.traces.map((trace) => trace.name)
-    const tracesData = useTracesData(project.id, traceNames)
-    const [selectedTableCohort, setSelectedTableCohort] = useState(null)
-    const [columns, setColumns] = useState([])
-    const [tableData, setTableData] = useState([])
+    const traceNames = table.traces.map((trace) => trace.name);
+    const tracesData = useTracesData(project.id, traceNames);
+    const [selectedTableCohort, setSelectedTableCohort] = useState(null);
+    const [columns, setColumns] = useState([]);
+    const [tableData, setTableData] = useState([]);
+
+    const csvConfig = mkConfig({
+        fieldSeparator: ',',
+        decimalSeparator: '.',
+        useKeysAsHeaders: true,
+    });
 
     useEffect(() => {
         if (selectedTableCohort && tracesData) {
-            setColumns(tableColumnsWithDot(table, selectedTableCohort.data, selectedTableCohort.traceName))
+            setColumns(tableColumnsWithDot(table, selectedTableCohort.data, selectedTableCohort.traceName));
         }
     }, [selectedTableCohort, tracesData, table]);
 
     useEffect(() => {
         if (selectedTableCohort && columns) {
-            setTableData(tableDataFromCohortData(selectedTableCohort.data, columns))
+            setTableData(tableDataFromCohortData(selectedTableCohort.data, columns));
         }
     }, [selectedTableCohort, columns]);
+    const handleExportData = () => {
```

Comment:
Wonder if this can be tested.

Reply:
Added a test for clicking the download button!